### ISSUE TITLE
parallelism wrong in basicSampling.random_sample_set

### ISSUE THREAD

**Issue by smattis** [UT-CHG/BET#247]
The parallelism is wrong in `bet.basicSampling.random_sample_set` and maybe other places in basicSampling. This is causing mistakes downstream. For instance, running the following with two processors

```
import numpy as np
import bet.sampling.basicSampling as bsam

domain = np.array([[0.0, 1.0], [0.0, 1.0]])
s_set = bsam.random_sample_set('r', domain, 10, globalize=False)
print s_set._values_local.shape
```
I get
```
(10, 2)
(10, 2)
```
Instead of creating 5 local samples on each processor, it is creating 10 different ones on each. Even worse is the following. Doing this:
```
import numpy as np
import bet.sampling.basicSampling as bsam
from bet.Comm import comm

domain = np.array([[0.0, 1.0], [0.0, 1.0]])
s_set = bsam.random_sample_set('r', domain, 10, globalize=True)

if comm.rank == 0:
    print s_set._values
comm.barrier()
if comm.rank == 1:
    print s_set._values
```
on two processors, I get:
```
[[ 0.91447128  0.01849912]
 [ 0.7560347   0.15781932]
 [ 0.96420241  0.32642613]
 [ 0.78393288  0.56364787]
 [ 0.96266756  0.30402477]
 [ 0.93050806  0.74384968]
 [ 0.07707351  0.29510574]
 [ 0.88228676  0.75432659]
 [ 0.48451015  0.74400973]
 [ 0.08062014  0.51465719]]
[[ 0.59228108  0.35506499]
 [ 0.44998455  0.78283404]
 [ 0.22877134  0.93470719]
 [ 0.74740891  0.24861271]
 [ 0.30125231  0.32583288]
 [ 0.99447317  0.33127152]
 [ 0.11202964  0.31436313]
 [ 0.3588562   0.8812713 ]
 [ 0.22556888  0.35365655]
 [ 0.29418263  0.66382836]]
```
This means that the sample_set._values on each processor are different, which obviously messes everything up downstream and causes crazy errors. This needs to be fixed ASAP, as it is being used in examples. Also, were things like this done in other parts of sampling?

**Comment by lcgraham** [comment]
This is part of what I'm trying to fix in pull request #243.

**Comment by smattis** [comment]
Ok. I see that now.

**Comment by lcgraham** [comment]
This issue should be fixed once this pull request goes through but I'm having trouble figuring out what is wrong since it passes all the tests on the departmental machine I'm using at FSU. And yes it appears in the regular sampling method in basicSampling and should be fixed there once #243 is merged.
